_posts/2020-06-18-a-sustainable-nuget-marketplace-will-have-to-compete-with-the-nuget-gallery.markdown

```diff
@@ -0,0 +1,368 @@
+
+Morally, this is "not done", but legally, this may happen. The marketplace will have to prevent this and have
+rules and guidance on how to go about this. We want sustainability!
+
+Another similarity with open source foundations is in the legality of purchasing packages.
+A NuGet marketplace will have to make sure consumers who purchase here can trust their purchase is legal
+and they can use the package without issue, now and in the future.
+
+Others are GDPR (as a consumer, is my personal data safe?), and probably some more. Feel free to add them
+in the comments and I will update the post.
+
+### For producers: business services
+
+Before we can sell packages, what's the legal status of an open source project? Is it a company? A non-profit?
+
+A NuGet marketplace will probably be the legal entity where consumers purchase packages from.
+Much like how [Fastspring](https://www.fastspring.com/) acts as a reseller, a NuGet marketplace will have to
+do this as well.
+
+A NuGet marketplace will have to act as a reseller, for several reasons:
+
+* Consumers purchase from one legal entity, with one contract.
+* Consumers can have e.g. a subscription that includes paid-for open source from multiple vendors, without dealing with all of them separately.
+* Consumers will be in several locations - the marketplace will have to accomodate for Value Added Tax (VAT), the EU VAT MOSS, and all that.
+
+If we want to make the step to sustainable open source, a NuGet marketplace will have to help
+projects in doing so. Not simply by having a downloadable binary with a paywall, but by helping them out
+with being a business.
+
+### For producers and consumers: flexible licensing and purchasing options
+
+A NuGet marketplace will need to support various purchasing options. Some examples:
+
+* Trial versions (30 days of usage)
+* Subscriptions (containing one or multiple packages)
+* Licenses with # seats, ideally also where enterprises can manage who can consume packages
+* Perpetual licenses
+* Licenses that are valid for major versions of a package only, etc.
+
+And as a consumer, can we pay by credit card only? Raise a purchase order and have NET90 payment terms?
+
+### For producers: where does the money go?
+
+Since many open source projects don't have a legal entity, where do funds go? As [Khalid Abuhakmeh](https://twitter.com/buhakmeh/) mentioned:
+
+<blockquote class="twitter-tweet"><p lang="en" dir="ltr">I think the stickier topic is how are funds appropriated to contributors. At some point these projects grow passed their creators.</p>&mdash; Khalid (@buhakmeh) <a href="https://twitter.com/buhakmeh/status/1273220645346856960?ref_src=twsrc%5Etfw">June 17, 2020</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+When there are 4 core contributors, who gets the proceeds? This is one a NuGet marketplace could solve
```

Comment:
My take is - this is an issue that the software publishers are going to have to work out among themselves. Not really the marketplace's role. But, we'll have to publish some guidance educating OSS develoeprs on how to handle these types of matters - it's not much different from forming a business.